doc/deps.md

```diff
@@ -106,21 +158,13 @@ cd thrift-0.13.0
 
 # Ant is required for having Java support in Thrift.
 sudo apt-get install ant
```

Comment:
Maybe this is for me, but simply running the build on 18.04 does not work in 2020. I'm testing the build on a brand new virtual machine installed this morning.
Even though [Gradle started not to support HTTP download](https://blog.gradle.org/decommissioning-http) in 2020, Thrift 0.13 -- released 2019 Oct -- is still using an HTTP URL. The following output happens during the build, towards the end:

~~~
make[3]: Entering directory '/home/codecompass/thrift-0.13.0/lib/java'
./gradlew  assemble \
        -Prelease=true \
        -Pthrift.version=0.13.0 \
        --console=plain
> Task :copyDependencies FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':copyDependencies'.
> Could not resolve all files for configuration ':testRuntime'.
   > Could not resolve org.slf4j:slf4j-api:1.7.25.
     Required by:
         project :
      > Could not resolve org.slf4j:slf4j-api:1.7.25.
         > Could not get resource 'http://repo1.maven.org/maven2/org/slf4j/slf4j-api/1.7.25/slf4j-api-1.7.25.pom'.
            > Could not GET 'http://repo1.maven.org/maven2/org/slf4j/slf4j-api/1.7.25/slf4j-api-1.7.25.pom'. Received status code 501 from server: HTTPS Required
   > Could not resolve org.apache.httpcomponents:httpclient:4.5.6.
     Required by:
         project :
      > Could not resolve org.apache.httpcomponents:httpclient:4.5.6.
         > Could not get resource 'http://repo1.maven.org/maven2/org/apache/httpcomponents/httpclient/4.5.6/httpclient-4.5.6.pom'.
            > Could not GET 'http://repo1.maven.org/maven2/org/apache/httpcomponents/httpclient/4.5.6/httpclient-4.5.6.pom'. Received status code 501 from server: HTTPS Required
   > Could not resolve org.apache.httpcomponents:httpcore:4.4.1.
~~~

The relevant breaking line is in `lib/java/gradle.properties`:
~~~{.gradle}
mvn.repo=http://repo1.maven.org/maven2
~~~

Swapping HTTP to HTTPS in the Gradle cfg fixes this:

```suggestion
sudo apt-get install ant
# Fix Thrift install script not using HTTPS for Java dependencies (http:// -> https://)
sed 's/http:\/\//https:\/\//g' -i ./lib/java/gradle.properties
```

Reply:
I will check this in a Docker container on a fresh system soon.

Also, we are at this point of the guide, we don't need Ant to be installed. I don't have ant installed. The Docker environment does not install ant.

Reply:
My conclusion is that for CodeCompass we only need the **Java generator** and not the **Java library**, which we have also discussed through emails last November.

Without Ant the Java library won't be built and there is no *HTTP->HTTPS* issue with Gradle. (We could even add the `--without-java` flag to Thrift, actually the CI was building CodeCompass that way for quite a long time previously.)

Reply:
@mcserep Did you check this with running a server and trying out the search parser and search service? Those are our only upstream stuff that "depend on" Java, but I am not sure about their inside architecture.